.github/workflows/update-code-injection-models.yml

```diff
@@ -0,0 +1,72 @@
+name: 'Update code injection models'
+
+on:
+  # Run weekly (at arbitrary time)
+  schedule:
+    - cron: '31 7 * * 5'
+  workflow_dispatch:
+
+permissions: {}
+
+jobs:
+  update-models:
+    permissions:
+      contents: write
+      pull-requests: write
```

Comment:
If you prefer, this could be split into two jobs: One with only `read` privileges processing the CodeQL data, and one with `write` privileges which creates the pull request.

Reply:
Keeping these in the same job seems fine to me for now :slightly_smiling_face: